testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -1016,3 +1016,21 @@ def test_SegmentGroup_warn_getattr(self, u):
     def test_SegmentGroup_nowarn_getitem(self, u):
         with no_deprecated_call():
             u.segments[0]
+
+
+class TestAttributeSetting(object):
+    @pytest.mark.parametrize('groupname', ['atoms', 'residues', 'segments'])
+    def test_setting_group_fail(self, groupname):
+        u = make_Universe()
+
+        group = getattr(u, groupname)
+        with pytest.raises(AttributeError):
+            group.this = 'that'
+
+    @pytest.mark.parametrize('groupname', ['atoms', 'residues', 'segments'])
+    def test_setting_component_fails(self, groupname):
+        u = make_Universe()
+        component = getattr(u, groupname)[0]
+
+        with pytest.raises(AttributeError):
+            component.this = 'that'
```

Comment:
can you add a test that adding an attribute still works. Or at least point me to a test if it exists already.